notebooks/Notebook3.md

```diff
@@ -2,33 +2,51 @@
 
 ## Last week's critique
 
-**TODO:** Fill in this part with a summary and reflection on the critique you received for
-last week's work. Answer questions such as:  How, specifically, did the feedback help
-improve the project? Did the feedback point out or offer something you hadn't considered?
-Did it help you make a design decision? Was it helpful in addressing the most pressing
-issues in your project? How will you incorporate the feedback into your work? Will you
-change something about the design, implementation, or evaluation as a result?
+This week I worked more on the parser for the DSL. Most of the feedback received was to
+use a parsing library such as parsec.py. After talking with professor Wiedermann, I
+decided to just use Python's built in string processing. The input being taken in isn't
+anything too complex/it doesn't require parser combinators, and I can also throw error
+messages in case syntax is missing and point out where an error was thrown. In these
+error messages I also give examples of the ideal syntax, that way users are able to
+follow along in case the syntax/messaging is vague. 
 
 ## Description
 
-**TODO:** Fill in this part with information about your work this week:
-important design decisions, changes to previous decisions, open questions,
-exciting milestones, preliminary results, etc. Feel free to include images
-(e.g., a sketch of the design or a screenshot of a running program), links to
-code, and any other resources that you think will help clearly convey your
-design process.
+This week I worked more on the parser, and I also started work on the .ics library. I
+was able to output a .ics calendar using the information a user would save in a file!
+Most of the parser and the .ics output is complete, I just need to troubleshoot a bit
+more on some things that are giving me problems. I also worked on using the datetime
+library so that events could be made using the .ics library.
 
 ## Questions
 
-**What is the most pressing issue for your project? What design decision do
-you need to make, what implementation issue are you trying to solve, or how
-are you evaluating your design and implementation?**
+Currently the most pressing issue is the lack of RRULE support in the .ics
+library. RRULE allows recurring events in calendars, so an event could occur weekly.
+This isn't supported by the .ics library (which has been a feature requested since 2016,
+so I doubt it'll be added in soon), so I'm going to try to instead output the .ics file,
+open it into python once made as a plain text file, and add the RRULE manually. I
```

Comment:
If you do decide to go the route of saving the file without `RRULE`s and then reopening it to add them manually, I wonder if there's a way you could add in some sort of placeholders when you're first writing the file so you don't have to spend as much effort parsing the file to find where to add the `RRULE`s on the second pass. I'm not familiar with the domain, but is there a feature that your library *can* write that you're not using (e.g., sub-second time resolution) you can use to encode the spot at which a `RRULE` should be inserted? Or maybe the `.ics` format is simple enough that you don't need this.